### ISSUE TITLE
[2.6] JWT documentation broken

### ISSUE THREAD

**Issue by NicoHaase** [api-platform/docs#1272]
https://api-platform.com/docs/core/jwt/#adding-endpoint-to-swaggerui-to-retrieve-a-jwt-token contains code that only works on PHP 8. The `Model/PathItem` part contains named arguments, changing this to code compatible with PHP 7 would be great :) 

I have to figure out how to make it work before I can suggest an alternative

**Comment by alanpoulain** [comment]
Hello,
This is wanted. Maintaining a documentation with a support of old PHP versions is too much work, we prefer to consider you are using the latest stable version of PHP.

**Comment by NicoHaase** [comment]
Don't get me wrong, I can only imagine the nightmare of supporting versions that differ so much, but couldn't you make `api-platform/core` v2.6 also dependent on PHP 8 only if you don't want to provide documentation about how to get it work on PHP 7.2, which is the currently lowest version supported?

**Comment by dunglas** [comment]
The API Platform documentation assumes that you use the latest version of the API Platform distribution, which is shipped with PHP 8. For clarity and to promote best practices, we always use in the docs the versions of our dependencies (PHP, Symfony...) shipped with the distribution.

At code level, API Platform support ms older versions of its dependencies, but it's mostly to ease the upgrade process. This allows upgrading package by package (start by upgrading PHP, then Symfony, then API Platform). However, we recommend to always upgrade your dependencies before starting to work on new features.

We will **not** document how to use the latest version of API Platform with old versions of its dependencies. This would require too much work, be a maintenance nightmare and would have almost no benefits.

TLDR: using the latest version of API Platform with non-latest dependencies is supported (and automatically tested) but we always assume you're using the latest versions in the documentation. If you want to mix old dependencies with new versions of API Platform, you're on your own (but it's usually easy to adapt the documentation for older versions).

Supporting the latest versions is already very tedious. We don't have the resource to document all possible dependency mixes and I don't think that it's a good idea anyway.
